lib/commands/touch.js

```diff
@@ -1,9 +1,53 @@
 const commands = {};
-
 //This is needed to make clicks on -image elements work properly
 commands.performActions = async function (actions) {
-  return await this.winAppDriver.sendCommand('/actions', 'POST', {actions});
+  switch (actions[0].id) {
+    case 'default keyboard':
+      this.log.info('Mapping keyboard actions');
+      return await performKeyboardCommands.bind(this)(actions[0].actions);
+    case 'default mouse':
+      this.log.info('Mapping mouse actions');
+      return await performMouseCommands.bind(this)(actions[0].actions);
+    default:
+      this.log.info('Going default actions route');
+      return await this.winAppDriver.sendCommand('/actions', 'POST', {actions});
+  };
 };
 
+async function performKeyboardCommands(actions) {
+  const keysFormatted = formatKeyboardInput(actions);
+  return await this.winAppDriver.sendCommand('/keys', 'POST', {'value': keysFormatted });
+}
+
+async function performMouseCommands(actions) {
+  let result;
+  for (const action in actions) {
+    switch (actions[action].type) {
+      case 'pointerMove':
+        result = await this.winAppDriver.sendCommand('/moveto', 'POST', {'element': actions[action].origin.ELEMENT });
+        break;
+      case 'pointerDown':
+        result = await this.winAppDriver.sendCommand('/buttondown', 'POST', {'button': actions[action].button });
```

Comment:
button may be undefined

Reply:
The click comes from an action itself, the interface forces you to specify the button itself for all the actions I found.
I can add some small logic there to have an empty body if the button is undefined but it seems redundant a bit to me

Reply:
you could just do it like `actions[action].button ??  0`, which would always default to the left mouse button

Reply:
fair enough, changed it